Relay Mode

olliw42 opened the pull request:
for those who want to play and test :)

this adds the new option "crsf tx" to the RxOutMode parameter. This option should be selected on the receiver on the relay station (= secondary receiver). This is, in principle, all as regards of the required configuration.

However, in addition some configurations need to be done to have it all working together, e.g. also the baudrate should be selected to match the tx module on the relay station (= main tx module), usually 115200. Also RxChOrder should be selected to match that in the tx module in the radio (= secondary tx module). "crsf" should be selected in both tx modules, and mavlink must be enabled in all devices.

The out pin of the secondary receiver should be connected to the jrpin5/in pin on the main tx module, as well as the serials. With regards to the serials, it is mandatory to connect the tx line on the secondary receiver, the rx line can be left open, but of cousre also be connected if full mavlink shall be relayed.

The RC data is properly transmitted, also the mavlink data, and yappu works on the radio as well as the crsf telemetry shows the stats corresponding to the main link, not the relay link.

The main tx module and main receiver cannot be configured via the lua script. That is for configuring the main link, and easy approach would be to just plug the main tx module into a radio and set things up for them.

rotorman commented:
Here two images that might help to understand how the relay mode could be used (just showing two examples, more constellations are possible):

![grafik](https://github.com/olliw42/mLRS/assets/21011587/2fabc504-3d0e-41a9-b8ad-c05f1d2d909c)

![grafik](https://github.com/olliw42/mLRS/assets/21011587/65bd4e97-2158-4ee6-9d6b-3a2b5735166d)

jiri-jirus commented:
I am here to test relay mode. would it be possible to prepare up-to-date firmwares for matek 2.4&868 RX/TX please? thanks

olliw42 committed:
common,tx,rx: tidies, tiny stuff in cli, mav rx

olliw42 committed:
tx, yield 100% LQ upon first connect

olliw42 committed:
tx,rx, MavlinkSystemID setup parameter

olliw42 committed:
Merge branch 'dev-matek-mtx' into merge-main

# Conflicts:
#	mLRS/CommonTx/crsf_interface_tx.h

olliw42 committed:
readme, update link to flasher

Updated links for mLRS Flasher applications.

olliw42 committed:
esp, correct dtr,rts handling

olliw42 committed:
wireless bridge, slight improvement

olliw42 committed:
readme, few tidies

olliw42 committed:
wireless bridge, tiny tidy

olliw42 committed:
esp, support usb dtr rts

olliw42 committed:
common, setup,common tidies

olliw42 committed:
Merge branch 'dev-matek-mtx' into merge-main

# Conflicts:
#	mLRS/Common/common.h
#	mLRS/Common/setup.h

olliw42 committed:
common, setup, allow for no serial, only serial2

olliw42 committed:
Merge branch 'dev-matek-mtx' into merge-main

olliw42 committed:
hal, mtx-db: move BT/ESP to serial2, activate esp configure

olliw42 committed:
prevent 868+24 and 915+24 to connect

olliw42 committed:
tx, simplify fhss index, band handling

olliw42 committed:
bind, tidy

olliw42 committed:
rx, bug with diversity setting (thx to wuyang521)

olliw42 committed:
rx, bug with power setting (thx to wuyang521)

jlpoltrack committed:
rx, msp, add Rx Reboot

* msp rx reboot

* feedback

olliw42 committed:
rx, msp, reboot tidy

olliw42 committed:
setup, correct a bug with rx diversity (wrong param used) and some tidies

olliw42 committed:
more bind

olliw42 committed:
setup, fhss, bind, tidies

olliw42 committed:
cc

olliw42 committed:
tx, bind, remove wrapper

olliw42 committed:
common, rename to SX_FHSS_FREQUENCY_BAND_XX

olliw42 committed:
bind, use frequency band send by tx

olliw42 committed:
common, simplify fhss num

olliw42 committed:
common, rename to BindScan_mask

olliw42 committed:
common, fhss, undo bind list, simplify to single bind channel

olliw42 committed:
common, some tidies

olliw42 committed:
common,rx,tx, add link_tx_status

olliw42 committed:
tx, cli, disp, improve for dualband

olliw42 committed:
common, hal, power, mtx

olliw42 committed:
update pre-release firmwares

jlpoltrack committed:
firmware, wireless, esp32c3 bridge update

olliw42 committed:
copy over mtx2

olliw42 committed:
Merge branch 'dev-matek-mtx' into merge-main

# Conflicts:
#	mLRS/Common/bind.h
#	mLRS/Common/fhss.h
#	mLRS/Common/setup.h
#	mLRS/Common/setup_types.cpp
#	mLRS/Common/setup_types.h
#	mLRS/Common/sx-drivers/sx128x_driver.h

olliw42 committed:
remove debugs

olliw42 committed:
update submodule dronecan_dsdlc

olliw42 committed:
common, dronecan, empy aka @) issue resolved with latest dronecan_dsdlc update

olliw42 committed:
Merge branch 'dev-matek-mtx' into merge-main

olliw42 committed:
Merge branch 'dev-relay' into merge-main

# Conflicts:
#	firmware/pre-release-esp/rx-bayck-nano-pro-900-v1.3.06.bin
#	firmware/pre-release-esp/rx-bayck-nano-pro-900-v1.3.07-@c6e0fef8.bin
#	firmware/pre-release-esp/rx-bayck-nano-pro-900-v1.3.07-dev-relay-@8e6112f1.bin
#	firmware/pre-release-esp/rx-betafpv-superd-2400-v1.3.07-dev-relay-@8e6112f1.bin
#	firmware/pre-release-esp/rx-generic-2400-d-pa-v1.3.06.bin
#	firmware/pre-release-esp/rx-generic-2400-d-pa-v1.3.07-@c6e0fef8.bin
#	firmware/pre-release-esp/rx-generic-2400-d-pa-v1.3.07-dev-relay-@8e6112f1.bin
#	firmware/pre-release-esp/rx-generic-2400-pa-v1.3.06.bin
#	firmware/pre-release-esp/rx-generic-2400-pa-v1.3.07-@c6e0fef8.bin
#	firmware/pre-release-esp/rx-generic-2400-pa-v1.3.07-dev-relay-@8e6112f1.bin
#	firmware/pre-release-esp/rx-generic-2400-td-pa-v1.3.07-dev-relay-@8e6112f1.bin
#	firmware/pre-release-esp/rx-generic-2400-v1.3.06.bin
#	firmware/pre-release-esp/rx-generic-2400-v1.3.07-@c6e0fef8.bin
#	firmware/pre-release-esp/rx-generic-2400-v1.3.07-dev-relay-@8e6112f1.bin
#	firmware/pre-release-esp/rx-generic-900-pa-v1.3.06.bin
#	firmware/pre-release-esp/rx-generic-900-pa-v1.3.07-@c6e0fef8.bin
#	firmware/pre-release-esp/rx-generic-900-pa-v1.3.07-dev-relay-@8e6112f1.bin
#	firmware/pre-release-esp/rx-generic-900-v1.3.06.bin
#	firmware/pre-release-esp/rx-generic-900-v1.3.07-@c6e0fef8.bin
#	firmware/pre-release-esp/rx-generic-900-v1.3.07-dev-relay-@8e6112f1.bin
#	firmware/pre-release-esp/rx-generic-lr1121-td-v1.3.07-dev-relay-@8e6112f1.bin
#	firmware/pre-release-esp/rx-radiomaster-br3-900-v1.3.06.bin
#	firmware/pre-release-esp/rx-radiomaster-br3-900-v1.3.07-@c6e0fef8.bin
#	firmware/pre-release-esp/rx-radiomaster-br3-900-v1.3.07-dev-relay-@8e6112f1.bin
#	firmware/pre-release-esp/rx-radiomaster-rp4td-2400-v1.3.07-dev-relay-@8e6112f1.bin
#	firmware/pre-release-esp/rx-radiomaster-xr4-900-v1.3.07-dev-relay-@8e6112f1.bin
#	firmware/pre-release-esp/rx-speedybee-nano-2400-v1.3.06.bin
#	firmware/pre-release-esp/rx-speedybee-nano-2400-v1.3.07-@c6e0fef8.bin
#	firmware/pre-release-esp/rx-speedybee-nano-2400-v1.3.07-dev-relay-@8e6112f1.bin
#	firmware/pre-release-esp/tx-jumper-internal-2400-v1.3.07-dev-relay-@8e6112f1.bin
#	firmware/pre-release-esp/tx-radiomaster-rp4td-2400-sik-telem-v1.3.07-dev-relay-@8e6112f1.bin

olliw42 committed:
update pre-release

olliw42 committed:
tools, run_setup more cmd options

olliw42 committed:
rx, esp, hal, XR4 info on pin usage (nfc)

olliw42 committed:
esp, rx, xr4, ups (nfc)

olliw42 committed:
tx, jrpin5, bandage for EdgeTx H7 sport issue

jlpoltrack committed:
esp,rx, add Flysky PR02-2400 receiver

olliw42 committed:
update pre-release firmwares

olliw42 committed:
lua, index test script obsolete, request-response used in all luas now

jlpoltrack committed:
rx, msp, listener for bind

* msp add bind

* feedback

olliw42 committed:
common, msp, tidies (nfc)

olliw42 committed:
tools, silently unsupport tx R9MX, rx WioE5 Groove

olliw42 committed:
rx, update core files for matek 30c, td30

olliw42 committed:
common, code review hardenings, and a real bug

olliw42 committed:
common, sx drivers, tidy, remove timeout form SetToRx()

jlpoltrack committed:
esp32, undef MIN MAX to get rid of warnings

olliw42 committed:
rx, mavlink, tidies

olliw42 committed:
es, glue, tidy (nfc)

olliw42 committed:
esp, gue, get it right

jlpoltrack committed:
wireless bridge, check of arduino version for esp32c3

olliw42 committed:
lua, param load status, add number of read params

olliw42 committed:
lua, tidy

olliw42 committed:
lua, Ethos lua script, init

jlpoltrack committed:
common, rf power switch, linear mapping of power levels

* linear power levels

* feedback

* comment update

olliw42 committed:
update submodule DSDL

olliw42 committed:
rx, dronecan interface, update rc

olliw42 committed:
common,tx, ups, it's mbridge (nfc)

olliw42 committed:
es, common, hal, tidies of lr1121 comments (nfc)

olliw42 committed:
modules,common, dronecan driver and lib updates

olliw42 committed:
core, stm32g4 can devices should run on 160MHz and 80MHz can

olliw42 committed:
rx, dronecan interface update

olliw42 committed:
hal, matek can devices, 1k for debug

olliw42 committed:
rx, dronecan, datalen bug

olliw42 committed:
update submodule mavlink

olliw42 committed:
update submodule fastmavlink

olliw42 committed:
common,rx,tx: mavlink updates

olliw42 committed:
rx, mavlink, set ack result where it should be set, + few more

olliw42 committed:
rx, dronecan, get tt stats right

olliw42 committed:
common, mavlinkX, proper bit buffer, memset, + comments

olliw42 committed:
wireless bridge, massive rework, make better use of C++ and OOP

olliw42 committed:
rx, matex mrxx-30, core, to support CAN feature

olliw42 committed:
update pre-release firmwares

olliw42 committed:
rx, mavlink, slight imprevment for mlrs radio link stats messages

olliw42 committed:
dito

olliw42 committed:
rx, cli, bug, hc04 not shown

olliw42 committed:
common,, tx, set USE_ESP_WIFI_BRIDGE_CONFIGURE and use it consistently

olliw42 committed:
tx, cli, hc04 add option to get pin

olliw42 committed:
release notes, more docs

brad112358 committed:
Minor dissapearing cursor bug fix and some code cleanup for BW Lua (#405)

* Fix bug where we didn't start with the cursor on next and there was an invisible cursor location at the end of parameters.
Eliminate some unneeded code now that we have the same number of cursor index positions now on all pages except 99.

* Update compiled version

olliw42 committed:
wireless bridge, add  AT_PSWD, AT_NETSSID

olliw42 committed:
tx, esp, cli, cli comamnds to get&set pswd and net ssid

olliw42 committed:
lua, show max param number on reload

jlpoltrack committed:
wireless bridge, firmware update

* c3 bridge update

* latest bridge firmware

* readme too

olliw42 committed:
lua, little bug

olliw42 committed:
lua, tidiy READMEs

olliw42 committed:
tx, cli, esp, tasks, a bit more hardening

olliw42 committed:
common, tasks tidy (nfc)

olliw42 committed:
tx, cli, no systemboot for esp devices

olliw42 committed:
tx, cli, ups

olliw42 committed:
lua, rewrite using tables, add flash esp, make download counter work for etx

olliw42 committed:
v1.3.08 (pre-release)

olliw42 committed:
tools, web, adjust url to v1.3.08

olliw42 committed:
v bump v1.3.09 (=dev)

olliw42 committed:
Merge branch 'dev-relay' into merge-main

# Conflicts:
#	firmware/pre-release-esp/rx-bayck-nano-pro-900-v1.3.07-@c6e0fef8.bin
#	firmware/pre-release-esp/rx-bayck-nano-pro-900-v1.3.07-dev-relay-@300b95ec.bin
#	firmware/pre-release-esp/rx-bayck-nano-pro-900-v1.3.08.bin
#	firmware/pre-release-esp/rx-betafpv-superd-2400-v1.3.07-@c6e0fef8.bin
#	firmware/pre-release-esp/rx-betafpv-superd-2400-v1.3.07-dev-relay-@300b95ec.bin
#	firmware/pre-release-esp/rx-betafpv-superd-2400-v1.3.08.bin
#	firmware/pre-release-esp/rx-generic-2400-d-pa-v1.3.07-@c6e0fef8.bin
#	firmware/pre-release-esp/rx-generic-2400-d-pa-v1.3.07-dev-relay-@300b95ec.bin
#	firmware/pre-release-esp/rx-generic-2400-d-pa-v1.3.08.bin
#	firmware/pre-release-esp/rx-generic-2400-pa-v1.3.07-@c6e0fef8.bin
#	firmware/pre-release-esp/rx-generic-2400-pa-v1.3.07-dev-relay-@300b95ec.bin
#	firmware/pre-release-esp/rx-generic-2400-pa-v1.3.08.bin
#	firmware/pre-release-esp/rx-generic-2400-td-pa-v1.3.07-dev-relay-@300b95ec.bin
#	firmware/pre-release-esp/rx-generic-2400-td-pa-v1.3.08.bin
#	firmware/pre-release-esp/rx-generic-2400-v1.3.07-@c6e0fef8.bin
#	firmware/pre-release-esp/rx-generic-2400-v1.3.07-dev-relay-@300b95ec.bin
#	firmware/pre-release-esp/rx-generic-2400-v1.3.08.bin
#	firmware/pre-release-esp/rx-generic-900-pa-v1.3.07-@c6e0fef8.bin
#	firmware/pre-release-esp/rx-generic-900-pa-v1.3.07-dev-relay-@300b95ec.bin
#	firmware/pre-release-esp/rx-generic-900-pa-v1.3.08.bin
#	firmware/pre-release-esp/rx-generic-900-td-pa-v1.3.07-@c6e0fef8.bin
#	firmware/pre-release-esp/rx-generic-900-td-pa-v1.3.07-dev-relay-@300b95ec.bin
#	firmware/pre-release-esp/rx-generic-900-td-pa-v1.3.08.bin
#	firmware/pre-release-esp/rx-generic-900-v1.3.07-@c6e0fef8.bin
#	firmware/pre-release-esp/rx-generic-900-v1.3.07-dev-relay-@300b95ec.bin
#	firmware/pre-release-esp/rx-generic-900-v1.3.08.bin
#	firmware/pre-release-esp/rx-generic-c3-lr1121-v1.3.07-dev-relay-@300b95ec.bin
#	firmware/pre-release-esp/rx-generic-lr1121-td-v1.3.08.bin
#	firmware/pre-release-esp/rx-radiomaster-br3-900-v1.3.07-@c6e0fef8.bin
#	firmware/pre-release-esp/rx-radiomaster-br3-900-v1.3.07-dev-relay-@300b95ec.bin
#	firmware/pre-release-esp/rx-radiomaster-br3-900-v1.3.08.bin
#	firmware/pre-release-esp/rx-radiomaster-rp4td-2400-v1.3.07-@c6e0fef8.bin
#	firmware/pre-release-esp/rx-radiomaster-rp4td-2400-v1.3.07-dev-relay-@300b95ec.bin
#	firmware/pre-release-esp/rx-radiomaster-xr1-v1.3.07-dev-relay-@300b95ec.bin
#	firmware/pre-release-esp/rx-radiomaster-xr4-v1.3.07-@c6e0fef8.bin
#	firmware/pre-release-esp/rx-radiomaster-xr4-v1.3.07-dev-relay-@300b95ec.bin
#	firmware/pre-release-esp/rx-radiomaster-xr4-v1.3.08.bin
#	firmware/pre-release-esp/rx-speedybee-nano-2400-v1.3.07-@c6e0fef8.bin
#	firmware/pre-release-esp/rx-speedybee-nano-2400-v1.3.07-dev-relay-@300b95ec.bin
#	firmware/pre-release-esp/rx-speedybee-nano-2400-v1.3.08.bin
#	firmware/pre-release-esp/tx-jumper-internal-2400-v1.3.07-dev-relay-@300b95ec.bin
#	firmware/pre-release-esp/tx-jumper-internal-900-v1.3.07-dev-relay-@300b95ec.bin
#	firmware/pre-release-esp/tx-radiomaster-bandit-900-v1.3.07-dev-relay-@300b95ec.bin
#	firmware/pre-release-esp/tx-radiomaster-internal-2400-v1.3.07-dev-relay-@300b95ec.bin
#	firmware/pre-release-esp/tx-radiomaster-internal-boxer-2400-v1.3.07-dev-relay-@300b95ec.bin
#	firmware/pre-release-esp/tx-radiomaster-internal-tx15-v1.3.07-dev-relay-@300b95ec.bin
#	firmware/pre-release-esp/tx-radiomaster-rp4td-2400-sik-telem-v1.3.07-@c6e0fef8.bin
#	firmware/pre-release-esp/tx-radiomaster-rp4td-2400-sik-telem-v1.3.07-dev-relay-@300b95ec.bin
#	firmware/pre-release-esp/tx-radiomaster-rp4td-2400-sik-telem-v1.3.08.bin
#	firmware/pre-release-stm32/rx-R9M-f103c8-elrs-bl-v1.3.07-dev-relay-@300b95ec.elrs
#	firmware/pre-release-stm32/rx-R9MM-f103rb-elrs-bl-v1.3.07-dev-relay-@300b95ec.elrs
#	firmware/pre-release-stm32/rx-R9MX-l433cb-elrs-bl-v1.3.07-dev-relay-@300b95ec.elrs
#	firmware/pre-release-stm32/tx-R9MX-l433cb-elrs-bl-v1.3.07-dev-relay-@300b95ec.elrs
#	lua/mLRS.lua
#	mLRS/Common/mavlink/olliw_dev.xml

olliw42 committed:
precompiled firmwares

olliw42 committed:
ups (nfc)

jlpoltrack committed:
lua, parameter download indicator, 320 fix

* lua parameter load 320 fix

* simpler

* olli snippet, extra spaces

olliw42 committed:
lua, slight param counter improvement, few tidies

olliw42 committed:
tx, mbridge, clean device info

olliw42 committed:
update submodule std32ll-lib

jlpoltrack committed:
disp, add DISP_CONTENT_Y_BASE

* disp y offset

* Add conditional definition for DISP_CONTENT_Y_BASE

* Align defines

olliw42 committed:
tx, cli, esp, info, extend delay, add espname

olliw42 committed:
rx, mavlink, tidy

olliw42 committed:
common, sx drivers, Config.FrameSyncWord to Sx.FlrcSyncWord

olliw42 committed:
sx drivers, intify sx_power, tidies (gain, max)

olliw42 committed:
sx-drivers, tidies (_MAX_DBM to _MAX, comments)(nfc)

olliw42 committed:
sx-driver, lr11xx reg to freq corrected (bug)

olliw42 committed:
update submodules stm32ll-lib, sx12xx-lib

olliw42 committed:
sx128x-driver, remove crcseed

jlpoltrack committed:
modules, stm32_usb_device, add getbuf

olliw42 committed:
Merge branch 'main' of https://github.com/olliw42/mLRS

olliw42 committed:
sx-drivers, rename gfsk br_bps to BitRate_bps

brad112358 committed:
wireless, fix UDP STA bug, empty password variable for default

robthomson committed:
Align ethos lua to new 1.7 release that is called Ethos 26.1.0

jlpoltrack committed:
wireless bridge, add ESPNow feature

* Add ESPNow to Bridge

* No LR Mode

* Binaries

* suggestions

* date, binaries

* no board

* GCS, fixed WiFi Channel for MSP

* No static, variable rename, leds.h, RGB led

* olli ring buffer optimization

* library tidy

olliw42 committed:
wireless-bridge, non-blocking setup, streamline, tidies

olliw42 committed:
add rx mr24-30c target, undo mRxx-30-can feature

olliw42 committed:
rx, mr24-30c, mr900-30c, mr900-td30, change rx_ser_port default to can

olliw42 committed:
update prerelease firmware

olliw42 committed:
rx, matek, it is 30td not td30

olliw42 committed:
update prerelease firmwares

olliw42 committed:
tx, esp,disp, update info after cli commands, show info on extra oled page

olliw42 committed:
ups

olliw42 committed:
wireless, correct name for mTX-DB30 (nfc), and some more cleanify

olliw42 committed:
wireless, consistent check for targets & cores, expect 3.3.7 on eps32

olliw42 committed:
wireless, tidy up espnow somewhat, should be nfc

jlpoltrack committed:
wireless bridge, correct channel 13 enable for esp32

jlpoltrack committed:
wireless bridge, bug, use eFuse Mac for ESP32-C3

olliw42 committed:
wireless bridge, tidies

olliw42 committed:
common, remove legacy stuff

olliw42 committed:
lua, color, for etx, drop a CRSF 400k warning when no communication

jlpoltrack committed:
firmware, wireless ESP32C3 binary update

olliw42 committed:
firmware, wireless ESP8266 binary update

olliw42 committed:
tx,common, mbridge, tidies (nfc)

olliw42 committed:
tools, run make firmwares, handle 00 versions correctly

olliw42 committed:
tools, web, adjust url to v1.4.00

olliw42 committed:
v bump v1.4.01 (=dev)

olliw42 committed:
rx,tx,common, new dual band, to also allow multiband

olliw42 committed:
tx, mtx db30

olliw42 committed:
remove five legacy targets (FRM303, 2x diy, tx-R9MX)

olliw42 committed:
tools, remove submodules for f0

olliw42 committed:
common,rx,tx, remove USE_FEATURE_FLRC, USE_FEATURE_MAVLINKX (yeah)

olliw42 committed:
common,rx, misc tidies (nfc)

olliw42 committed:
common,rx,tx,modules, move esp-lib, stdstm32-can to modules

olliw42 committed:
add lr2021

olliw42 committed:
tools, more remove f0

olliw42 committed:
update pre-compiled binaries

olliw42 committed:
wireless bridge, rename file to include chip

jlpoltrack committed:
common, sx drivers, 128x, 126x, remove delays

olliw42 committed:
common, sx drivers, assume 0 buffer pointer in read frame

jlpoltrack committed:
common, sx driver, lr1121, has no buffer base command

olliw42 committed:
common, sx-drivers, lr1121, comment concerning base address

jlpoltrack committed:
wireless bridge, core 3.x for esp32c3, also esp32

* Bridge 3.X Again

* Add Firmware

* Cleanup

jlpoltrack committed:
esp, esp-lib, uart, better bytesavailable

* esp uart better bytesavailable

* feedback

olliw42 committed:
common,tx, deprecated/remove CMD_PARAM_REQUEST_LIST

jlpoltrack committed:
sx-driver, lr202, bug in DIO RF config

* LR2021 DIO RF Config

* Mask back

jlpoltrack committed:
esp, esp-lib,disp,tx, I2C on core 0, no spin

* esp i2c core 0

* Tidy

* cmdhome every time

jlpoltrack committed:
esp, rx, rxclock with one-shot timers

* esp rxclock

* esp8266 underflow catch

* cleared ifdef elif

* Tidy

jlpoltrack committed:
esp, tx, jrpin5 half duplex switching task

* esp half duplex task

* full duplex guard

jlpoltrack committed:
esp, esp-lib, I2C comment (nfc)

jlpoltrack committed:
esp, esp-idf, pio update

olliw42 committed:
esp, framework update followup, exclude I2S to drop lots of warnings

jlpoltrack committed:
sx-driver, lr20xx, HF PaConfig mask fix

b14ckyy committed:
lua, ethos, fix list param selection (allowed_mask filtering, immediate value update)

- buildChoicesFromOptions now filters by allowed_mask so only firmware-allowed options are shown
- Setter updates p.value immediately before async commitParam, preventing getter from reverting to old value
- Removed incorrect getter clamping that mapped firmware indices to visible-choice indices
- Applies to LIST, STR6, and numeric param types

Co-authored-by: b14ckyy <marc.b.hoffmann@live.de>

olliw42 committed:
lua, ethos, add some version timestamp (nfc)

jlpoltrack committed:
common, sx-drivers, lr20xx driver, add define to set TCXO

olliw42 committed:
common, sx-drivers, lr20xx driver, comment tidies (nfc)

jlpoltrack committed:
esp, add Radiomaster TX16S MK3 support (internal LR1121)

* add tx16s mk3 support

* Fix Compile Error

* Fixed constant name

---------

Co-authored-by: Tom McAdam <mcadam1980@gmail.com>

olliw42 committed:
common, sx driver, lr20xx driver, tidy comment, change to SX_USE_TCXO_VOLTAGE

olliw42 committed:
update pre-compiled binaries, to bring in RM TX16S MK3

jlpoltrack committed:
esp, platformi, easy internal tx flash using py script for passthru

* ESP Easy Internal TX Flash

* Add MK3 Support

* File Moved

jlpoltrack committed:
common, sx-drivers, lr2021, more 

* More LR Driver

* Configuration Defines

* No Braces, Names

moosss committed:
rx, dronecan, tools, use sys.executable to invoke dronescan_dsdlc.py

On systems where only python3 is available (not python), calling dronescan_dsdlc.py directly relies on its shebang (#!/usr/bin/env python) which fails. Using sys.executable ensures the same interpreter that is running dronecan_generate_c_library.py is used.

olliw42 committed:
rx, dronecan, tools, tidy dronescan generate py

olliw42 committed:
common,rx,tx, some include tidies (nfc)

olliw42 committed:
submodule update sx12xx-lib

olliw42 committed:
common, sx drivers, use module

olliw42 committed:
tx, remove some FRM left overs (nfc)

olliw42 committed:
tx, jr_pin5_interface, pin5_rx/tx_enable() to align with esp lib (nfc)

jlpoltrack committed:
esp, pio T15 Pro supported (nfc)

jlpoltrack committed:
esp, jumper internal 2.4 hal, T15 Pro supported (nfc)

jlpoltrack committed:
esp, jumper internal 900 hal, T15 Pro supported (nfc)

olliw42 committed:
tx, crsf/mbridge cmd timing handling improved

olliw42 committed:
tx, crsf, send link statistics in one crsf frame

olliw42 committed:
tx, jrpin5, stm32, 100us too short for H7

olliw42 committed:
tx,crsf, harden crsf parser

olliw42 committed:
tx, crsf, a bit more

olliw42 committed:
common, hal, generally provide delay functions

olliw42 committed:
common, sx drivers, tidies

olliw42 committed:
tx, crsf gps time, few more sensors, define CRSF_FRAME_BUNDLING, also bundle sensors

olliw42 committed:
uuppss

olliw42 committed:
tx,common, mbridge, a bit of tidy

olliw42 committed:
cc

olliw42 committed:
modules, usb device lib, tidy (nfc)

brad112358 committed:
lua, changes to support AX12 from tmcadam with workaround for missing getRSSI()

olliw42 committed:
lua, few tidies

olliw42 committed:
common, hal, for rx replace HAS_SERIAL_OR_DEBUG by HAS_NO_SERIAL, few more tidies

olliw42 committed:
common, hal, tidies (nfc)

olliw42 committed:
esp, hal, pr02, ups, filename corrected (nfc)

olliw42 committed:
ups

olliw42 committed:
common,rx,tx, rewrite serial ports asignments using pointers

olliw42 committed:
modules, usb-device lib, add protector (by jlp)

jlpoltrack committed:
tools, run_make_firmwares, glob dronecan out/src sources

jlpoltrack committed:
tools, run_make_firmwares, allow arm-none-eabi toolchain on PATH (no CubeIDE)

* tools, run_make_firmwares, arm-none-eabi toolchain on PATH (no CubeIDE)

* Feedback

* Update Date

* Nicer Print

* Redo

olliw42 committed:
tools, run_make_firmwares, tidies (nfc)

olliw42 committed:
esp, tx, tSerialBase needs full override (bug)

brad112358 committed:
wireless bridge, don't require correct serial port selection for wireless bridge flashing

olliw42 committed:
tx, hc04, streamline serial handling

jlpoltrack committed:
tools, run_make_firmwares, parallel builds

brad112358 committed:
common, tx, add RadioMaster AX12 target

* Add RadioMaster AX12 target

* Drop SERIAL_FIXED_BAUD in favor of directly checking for AX12

---------

Co-authored-by: JLP <jlpoltrack@gmail.com>

olliw42 committed:
common, tx, AX12 target tidies

olliw42 committed:
common, channel order, ups tidiy (nfc)

jlpoltrack committed:
tools, add py script for flashing on linux/macos

* stm32 flash tool

* rename

olliw42 committed:
tools, run flash stm32, tidies (nfc)

olliw42 committed:
update prerelease firmwares

olliw42 committed:
Merge branch 'dev-relay' into merge-main

# Conflicts:
#	firmware/pre-release-esp/rx-bayck-nano-pro-900-v1.3.08-dev-relay.bin
#	firmware/pre-release-esp/rx-bayck-nano-pro-900-v1.3.08.bin
#	firmware/pre-release-esp/rx-bayck-nano-pro-900-v1.4.01-@3a70706e.bin
#	firmware/pre-release-esp/rx-betafpv-superd-2400-v1.3.08-dev-relay.bin
#	firmware/pre-release-esp/rx-flysky-pr02-2400-td-esp32s3-v1.3.08-dev-relay.bin
#	firmware/pre-release-esp/rx-generic-2400-d-pa-v1.3.08-dev-relay.bin
#	firmware/pre-release-esp/rx-generic-2400-d-pa-v1.3.08.bin
#	firmware/pre-release-esp/rx-generic-2400-d-pa-v1.4.01-@3a70706e.bin
#	firmware/pre-release-esp/rx-generic-2400-pa-v1.3.08-dev-relay.bin
#	firmware/pre-release-esp/rx-generic-2400-pa-v1.3.08.bin
#	firmware/pre-release-esp/rx-generic-2400-pa-v1.4.01-@3a70706e.bin
#	firmware/pre-release-esp/rx-generic-2400-td-pa-v1.3.08-dev-relay.bin
#	firmware/pre-release-esp/rx-generic-2400-v1.3.08-dev-relay.bin
#	firmware/pre-release-esp/rx-generic-2400-v1.3.08.bin
#	firmware/pre-release-esp/rx-generic-2400-v1.4.01-@3a70706e.bin
#	firmware/pre-release-esp/rx-generic-900-pa-v1.3.08-dev-relay.bin
#	firmware/pre-release-esp/rx-generic-900-pa-v1.3.08.bin
#	firmware/pre-release-esp/rx-generic-900-pa-v1.4.01-@3a70706e.bin
#	firmware/pre-release-esp/rx-generic-900-v1.3.08-dev-relay.bin
#	firmware/pre-release-esp/rx-generic-900-v1.3.08.bin
#	firmware/pre-release-esp/rx-generic-900-v1.4.01-@3a70706e.bin
#	firmware/pre-release-esp/rx-generic-c3-lr1121-v1.3.08-dev-relay.bin
#	firmware/pre-release-esp/rx-generic-lr1121-td-v1.3.08-dev-relay.bin
#	firmware/pre-release-esp/rx-radiomaster-br3-900-v1.3.08-dev-relay.bin
#	firmware/pre-release-esp/rx-radiomaster-br3-900-v1.3.08.bin
#	firmware/pre-release-esp/rx-radiomaster-br3-900-v1.4.01-@3a70706e.bin
#	firmware/pre-release-esp/rx-radiomaster-rp4td-2400-v1.3.08-dev-relay.bin
#	firmware/pre-release-esp/rx-radiomaster-xr1-v1.3.08-dev-relay.bin
#	firmware/pre-release-esp/rx-radiomaster-xr4-v1.3.08-dev-relay.bin
#	firmware/pre-release-esp/rx-speedybee-nano-2400-v1.3.08-dev-relay.bin
#	firmware/pre-release-esp/rx-speedybee-nano-2400-v1.3.08.bin
#	firmware/pre-release-esp/rx-speedybee-nano-2400-v1.4.01-@3a70706e.bin
#	firmware/pre-release-esp/tx-jumper-internal-2400-v1.3.08-dev-relay.bin
#	firmware/pre-release-esp/tx-jumper-internal-900-v1.3.08-dev-relay.bin
#	firmware/pre-release-esp/tx-radiomaster-bandit-micro-900-v1.3.08-dev-relay.bin
#	firmware/pre-release-esp/tx-radiomaster-internal-2400-v1.3.08-dev-relay.bin
#	firmware/pre-release-esp/tx-radiomaster-internal-boxer-2400-v1.3.08-dev-relay.bin
#	firmware/pre-release-esp/tx-radiomaster-internal-gx12-v1.3.08-dev-relay.bin
#	firmware/pre-release-esp/tx-radiomaster-internal-tx15-v1.3.08-dev-relay.bin
#	firmware/pre-release-esp/tx-radiomaster-nomad-v1.3.08-dev-relay.bin
#	firmware/pre-release-esp/tx-radiomaster-ranger-2400-v1.3.08-dev-relay.bin
#	firmware/pre-release-esp/tx-radiomaster-rp4td-2400-sik-telem-v1.3.08-dev-relay.bin
#	firmware/pre-release-stm32/rx-R9M-f103c8-elrs-bl-v1.3.08-dev-relay.elrs
#	firmware/pre-release-stm32/rx-R9MM-f103rb-elrs-bl-v1.3.08-dev-relay.elrs
#	firmware/pre-release-stm32/rx-R9MX-l433cb-elrs-bl-v1.3.08-dev-relay.elrs
#	mLRS/CommonTx/crsf_interface_tx.h

jlpoltrack committed:
tx, hal, add in to db30

olliw42 committed:
common,tx, bugs in esp, msp, lots of tidies

olliw42 committed:
tx, crsf telemetry, add DISTANCE_SENSOR support

olliw42 committed:
tx, common, change TxSerDest parameter to TxSerPort

olliw42 committed:
common, sx-drivers, lr2021, correct pa config for 70 cm

olliw42 committed:
rx, dronecan, few tidies (nfc)

olliw42 committed:
rx, streamline extern declarations (nfc)

olliw42 committed:
tx, esp, uses extern Setup anyway, so streamline (nfc)

olliw42 committed:
tx, hc04, fix baud to 115k, and streamline

olliw42 committed:
wireless bridge, compile bug, move esp_wifi.h include (by Rothammel)

olliw42 committed:
common,tx,rx, msp updates

jlpoltrack committed:
common, setup, LR11xx, LR20xx, 433 MHz support

olliw42 committed:
common,tx,rx, serial rewrite (major changes)

olliw42 committed:
common, setup, 2.4 & 433,70

olliw42 committed:
remove some vaporware

olliw42 committed:
grr

olliw42 committed:
add diy nicerf lr2021 targets

olliw42 committed:
update prerelease firmwares

olliw42 committed:
common,tx, tidies, remove ° (nfc)

olliw42 committed:
Merge branch 'dev-relay' into merge-main

# Conflicts:
#	firmware/pre-release-esp/rx-bayck-nano-pro-900-v1.4.01-@3a70706e.bin
#	firmware/pre-release-esp/rx-bayck-nano-pro-900-v1.4.01-@6e44ed49.bin
#	firmware/pre-release-esp/rx-bayck-nano-pro-900-v1.4.01-merge-main-@da837a67.bin
#	firmware/pre-release-esp/rx-betafpv-superd-2400-v1.4.01-@6e44ed49.bin
#	firmware/pre-release-esp/rx-flysky-pr02-2400-td-esp32s3-v1.4.01-@3a70706e.bin
#	firmware/pre-release-esp/rx-flysky-pr02-2400-td-esp32s3-v1.4.01-@6e44ed49.bin
#	firmware/pre-release-esp/rx-flysky-pr02-2400-td-esp32s3-v1.4.01-merge-main-@da837a67.bin
#	firmware/pre-release-esp/rx-generic-2400-d-pa-v1.4.01-@3a70706e.bin
#	firmware/pre-release-esp/rx-generic-2400-d-pa-v1.4.01-@6e44ed49.bin
#	firmware/pre-release-esp/rx-generic-2400-d-pa-v1.4.01-merge-main-@da837a67.bin
#	firmware/pre-release-esp/rx-generic-2400-pa-v1.4.01-@3a70706e.bin
#	firmware/pre-release-esp/rx-generic-2400-pa-v1.4.01-@6e44ed49.bin
#	firmware/pre-release-esp/rx-generic-2400-pa-v1.4.01-merge-main-@da837a67.bin
#	firmware/pre-release-esp/rx-generic-2400-td-pa-v1.4.01-@6e44ed49.bin
#	firmware/pre-release-esp/rx-generic-2400-v1.4.01-@3a70706e.bin
#	firmware/pre-release-esp/rx-generic-2400-v1.4.01-@6e44ed49.bin
#	firmware/pre-release-esp/rx-generic-2400-v1.4.01-merge-main-@da837a67.bin
#	firmware/pre-release-esp/rx-generic-900-pa-v1.4.01-@3a70706e.bin
#	firmware/pre-release-esp/rx-generic-900-pa-v1.4.01-@6e44ed49.bin
#	firmware/pre-release-esp/rx-generic-900-pa-v1.4.01-merge-main-@da837a67.bin
#	firmware/pre-release-esp/rx-generic-900-td-pa-v1.4.01-@3a70706e.bin
#	firmware/pre-release-esp/rx-generic-900-td-pa-v1.4.01-@6e44ed49.bin
#	firmware/pre-release-esp/rx-generic-900-td-pa-v1.4.01-merge-main-@da837a67.bin
#	firmware/pre-release-esp/rx-generic-900-v1.4.01-@3a70706e.bin
#	firmware/pre-release-esp/rx-generic-900-v1.4.01-@6e44ed49.bin
#	firmware/pre-release-esp/rx-generic-900-v1.4.01-merge-main-@da837a67.bin
#	firmware/pre-release-esp/rx-generic-c3-lr1121-v1.4.01-@3a70706e.bin
#	firmware/pre-release-esp/rx-generic-c3-lr1121-v1.4.01-@6e44ed49.bin
#	firmware/pre-release-esp/rx-generic-c3-lr1121-v1.4.01-merge-main-@da837a67.bin
#	firmware/pre-release-esp/rx-generic-lr1121-td-v1.4.01-@3a70706e.bin
#	firmware/pre-release-esp/rx-generic-lr1121-td-v1.4.01-@6e44ed49.bin
#	firmware/pre-release-esp/rx-generic-lr1121-td-v1.4.01-merge-main-@da837a67.bin
#	firmware/pre-release-esp/rx-radiomaster-br3-900-v1.4.01-@3a70706e.bin
#	firmware/pre-release-esp/rx-radiomaster-br3-900-v1.4.01-@6e44ed49.bin
#	firmware/pre-release-esp/rx-radiomaster-br3-900-v1.4.01-merge-main-@da837a67.bin
#	firmware/pre-release-esp/rx-radiomaster-rp4td-2400-v1.4.01-@3a70706e.bin
#	firmware/pre-release-esp/rx-radiomaster-rp4td-2400-v1.4.01-@6e44ed49.bin
#	firmware/pre-release-esp/rx-radiomaster-rp4td-2400-v1.4.01-merge-main-@da837a67.bin
#	firmware/pre-release-esp/rx-radiomaster-xr1-v1.4.01-@3a70706e.bin
#	firmware/pre-release-esp/rx-radiomaster-xr1-v1.4.01-@6e44ed49.bin
#	firmware/pre-release-esp/rx-radiomaster-xr1-v1.4.01-merge-main-@da837a67.bin
#	firmware/pre-release-esp/rx-radiomaster-xr4-v1.4.01-@3a70706e.bin
#	firmware/pre-release-esp/rx-radiomaster-xr4-v1.4.01-@6e44ed49.bin
#	firmware/pre-release-esp/rx-radiomaster-xr4-v1.4.01-merge-main-@da837a67.bin
#	firmware/pre-release-esp/rx-speedybee-nano-2400-v1.4.01-@3a70706e.bin
#	firmware/pre-release-esp/rx-speedybee-nano-2400-v1.4.01-@6e44ed49.bin
#	firmware/pre-release-esp/rx-speedybee-nano-2400-v1.4.01-merge-main-@da837a67.bin
#	firmware/pre-release-esp/tx-betafpv-micro-1w-2400-v1.4.01-@3a70706e.bin
#	firmware/pre-release-esp/tx-betafpv-micro-1w-2400-v1.4.01-@6e44ed49.bin
#	firmware/pre-release-esp/tx-betafpv-micro-1w-2400-v1.4.01-merge-main-@da837a67.bin
#	firmware/pre-release-esp/tx-jumper-internal-2400-v1.4.01-@3a70706e.bin
#	firmware/pre-release-esp/tx-jumper-internal-2400-v1.4.01-@6e44ed49.bin
#	firmware/pre-release-esp/tx-jumper-internal-2400-v1.4.01-merge-main-@da837a67.bin
#	firmware/pre-release-esp/tx-jumper-internal-900-v1.4.01-@3a70706e.bin
#	firmware/pre-release-esp/tx-jumper-internal-900-v1.4.01-@6e44ed49.bin
#	firmware/pre-release-esp/tx-jumper-internal-900-v1.4.01-merge-main-@da837a67.bin
#	firmware/pre-release-esp/tx-radiomaster-bandit-900-v1.4.01-merge-main-@da837a67.bin
#	firmware/pre-release-esp/tx-radiomaster-bandit-micro-900-v1.4.01-merge-main-@da837a67.bin
#	firmware/pre-release-esp/tx-radiomaster-internal-2400-v1.4.01-merge-main-@da837a67.bin
#	firmware/pre-release-esp/tx-radiomaster-internal-boxer-2400-v1.4.01-merge-main-@da837a67.bin
#	firmware/pre-release-esp/tx-radiomaster-internal-gx12-v1.4.01-@6e44ed49.bin
#	firmware/pre-release-esp/tx-radiomaster-internal-tx15-v1.4.01-merge-main-@da837a67.bin
#	firmware/pre-release-esp/tx-radiomaster-internal-tx16smk3-v1.4.01-merge-main-@da837a67.bin
#	firmware/pre-release-esp/tx-radiomaster-nomad-v1.4.01-merge-main-@da837a67.bin
#	firmware/pre-release-esp/tx-radiomaster-ranger-2400-v1.4.01-merge-main-@da837a67.bin
#	firmware/pre-release-esp/tx-radiomaster-rp4td-2400-sik-telem-v1.4.01-@3a70706e.bin
#	firmware/pre-release-esp/tx-radiomaster-rp4td-2400-sik-telem-v1.4.01-@6e44ed49.bin
#	firmware/pre-release-esp/tx-radiomaster-rp4td-2400-sik-telem-v1.4.01-merge-main-@da837a67.bin
#	firmware/pre-release-stm32/rx-R9M-f103c8-elrs-bl-v1.4.01-@3a70706e.elrs
#	firmware/pre-release-stm32/rx-R9M-f103c8-elrs-bl-v1.4.01-@6e44ed49.elrs
#	firmware/pre-release-stm32/rx-R9M-f103c8-elrs-bl-v1.4.01-merge-main-@da837a67.elrs
#	firmware/pre-release-stm32/rx-R9MM-f103rb-elrs-bl-v1.4.01-@3a70706e.elrs
#	firmware/pre-release-stm32/rx-R9MM-f103rb-elrs-bl-v1.4.01-@6e44ed49.elrs
#	firmware/pre-release-stm32/rx-R9MM-f103rb-elrs-bl-v1.4.01-merge-main-@da837a67.elrs
#	firmware/pre-release-stm32/rx-R9MX-l433cb-elrs-bl-v1.4.01-@3a70706e.elrs
#	firmware/pre-release-stm32/rx-R9MX-l433cb-elrs-bl-v1.4.01-@6e44ed49.elrs
#	firmware/pre-release-stm32/rx-R9MX-l433cb-elrs-bl-v1.4.01-merge-main-@da837a67.elrs
#	mLRS/Common/theory_of_operation.h
#	mLRS/CommonTx/crsf_interface_tx.h
